### ISSUE TITLE
Search Chinese Content with matches, The result looks have some problem.

### ISSUE THREAD

**Issue by ldnvnbl** [meilisearch/meilisearch#2534]
**Describe the bug**

Search Chinese Content  with  match result, start index maybe not right.

**To Reproduce**

1. create index

[poems.json.zip](https://github.com/meilisearch/meilisearch/files/8933218/poems.json.zip)

```bash
curl \
  -X POST 'http://127.0.0.1:7700/indexes/poems/documents' \
  -H 'Content-Type: application/json' \
  --data-binary @poems.json
```

2. search with `"matches": true`

```bash
  curl -XPOST 'http://127.0.0.1:7700/indexes/poems/search' \
    -H 'Content-Type: application/json' \
    -d '{
            "q": "盈盈在紫微",
            "matches": true
        }' | jq
```

the result:

```json
{
  "hits": [
    {
      "id": "3",
      "title": "宫中行乐词八首",
      "authorName": "李白",
      "dynasty": "唐代",
      "type": 2,
      "content": "小小生金屋，盈盈在紫微。\n山花插宝髻，石竹绣罗衣。\n每出深宫里，常随步辇归。\n只愁歌舞散，化作彩云飞。\n\n柳色黄金嫩，梨花白雪香。\n玉楼巢翡翠，金殿锁鸳鸯。\n选妓随雕辇，征歌出洞房。\n宫中谁第一，飞燕在昭阳。\n\n卢橘为秦树，蒲桃出汉宫。\n烟花宜落日，丝管醉春风。\n笛奏龙吟水，箫鸣凤下空。\n君王多乐事，还与万方同。\n\n玉树春归日，金宫乐事多。\n后庭朝未入，轻辇夜相过。\n笑出花间语，娇来竹下歌。\n莫教明月去，留著醉嫦娥。\n\n绣户香风暖，纱窗曙色新。\n宫花争笑日，池草暗生春。\n绿树闻歌鸟，青楼见舞人。\n昭阳桃李月，罗绮自相亲。\n\n今日明光里，还须结伴游。\n春风开紫殿，天乐下朱楼。\n艳舞全知巧，娇歌半欲羞。\n更怜花月夜，宫女笑藏钩。\n\n寒雪梅中尽，春风柳上归。\n宫莺娇欲醉，檐燕语还飞。\n迟日明歌席，新花艳舞衣。\n晚来移彩仗，行乐泥光辉。\n\n水绿南薰殿，花红北阙楼。\n莺歌闻太液，凤吹绕瀛洲。\n素女鸣珠佩，天人弄彩球。\n今朝风日好，宜入未央游。",
      "_matchesInfo": {
        "content": [
          {
            "start": 18,   // I know use byte count it the start is right, but I expect start counted by Hanzi, if this the start is 7
            "length": 2   // this value looks count by Hanzi not byte
          },
          {
            "start": 24,
            "length": 1
          },
          {
            "start": 27,
            "length": 2
          },
          {
            "start": 284,
            "length": 1
          }
        ]
      }
    }
  ],
  "nbHits": 1,
  "exhaustiveNbHits": false,
  "query": "盈盈在紫微",
  "limit": 20,
  "offset": 0,
  "processingTimeMs": 12
}
```

**Expected behavior**

if the _matchesInfo result count by Hanzi, the first three position value should be merged, because they are continuous.

so I expected _matchesInfo result is:

```json

"_matchesInfo": {
    "content": [
        {
             "start": 7,
             "length": 5
        }
    ]
}

```

**Meilisearch version:** 

v0.27.2

mac os 12.4

**Comment by oluademola** [comment]
Hi @ldnvnbl 

Indeed It appears to be an issue on our side and should be fixed in v0.28 with the new tokenizer. 

Thanks for bringing this to our notice.

**Comment by curquiza** [comment]
I close this issue since it should be fixed with https://github.com/meilisearch/meilisearch/pull/2468

A pre-release (RC) is already available [here](https://github.com/meilisearch/meilisearch/releases/tag/v0.28.0rc0) if you want to test, or with the `v0.28.0rc0` docker image
